smarty/forcefield_utils.py

```diff
@@ -364,6 +364,67 @@ def getMolParamIDToAtomIndex( oemol, ff):
 
     return param_usage
 
+def merge_system( topology0, topology1, system0, system1, positions0, positions1, label0="AMBER system", label1 = "SMIRFF system", verbose = True):
+    """Merge two given OpenMM systems. Returns the merged OpenMM System.
+
+    Parameters
+    ----------
+    topology0 : OpenMM Topology
+        Topology of first system (i.e. a protein)
+    topology1 : OpenMM Topology
+        Topology of second system (i.e. a ligand)
+    system0 : OpenMM System
+        First system for comparison (usually from AMBER)
+    system1 : OpenMM System
+        Second system for comparison (usually from SMIRFF)
+    positions0 : simtk.unit.Quantity wrapped
+        Positions to use for energy evaluation comparison
+    positions1 (optional) : simtk.unit.Quantity wrapped (optional)
+        Positions to use for second OpenMM system
+    label0 (optional) : str
+        String labeling system0 for output. Default, "AMBER system"
+    label1 (optional) : str
+        String labeling system1 for output. Default, "SMIRFF system"
+    verbose (optional) : bool
+        Print out info on energies, True/False (default True)
+
+    Returns
+    ----------
+    topology : OpenMM Topology
+    system : OpenMM System
+    positions: unit.Quantity position array
+    """
+
+    #Load OpenMM Systems to ParmEd Structures
+    structure0 = parmed.openmm.load_topology( topology0, system0 )
+    structure1 = parmed.openmm.load_topology( topology1, system1 )
+
+    #Merge parameterized Structure
+    structure = structure0 + structure1
+    topology = structure.topology
+
+    #Concatenate positions arrays
+    positions0 = np.array(positions0._value)
```

Comment:
This is unsafe! If you want to convert to pure numpy arrays, use this idiom to safely manage units:
```python
positions_unit = unit.anstroms
positions0_dimensionless = positions0 / positions_unit
positions1_dimensionless = positions1 / positions_unit
# compute `result` somewhere in dimensionless units
positions = unit.Quantity(result_dimensionless, positions_unit)
```